pkg/catalog/catalog.go

```diff
@@ -7,38 +7,44 @@ import (
 )
 
 // Catalog provides an in-memory cache of profiles from the cluster which can be queried easily.
-type Catalog struct {
-	profiles []profilesv1.ProfileDescription
-}
+type Catalog map[string][]profilesv1.ProfileDescription
 
 // New creates a new, empty catalog.
-func New() *Catalog {
-	return &Catalog{profiles: []profilesv1.ProfileDescription{}}
+func New() Catalog {
+	return map[string][]profilesv1.ProfileDescription{}
 }
 
-// Add adds p profiles to the catalog.
-func (c *Catalog) Add(catalogName string, profiles ...profilesv1.ProfileDescription) {
-	for _, p := range profiles {
-		p.Catalog = catalogName
-		c.profiles = append(c.profiles, p)
+// Update updates the catalog by replacing existing profiles with new profiles
+func (c Catalog) Update(catalogName string, profiles ...profilesv1.ProfileDescription) {
+	for i := range profiles {
+		profiles[i].Catalog = catalogName
 	}
+	c[catalogName] = profiles
+}
+
+// Remove removes the specified catalog.
+func (c Catalog) Remove(catalogName string) {
+	delete(c, catalogName)
```

Comment:
```suggestion
// Remove removes the specified catalog source.
func (c Catalog) Remove(sourceName string) {
	delete(c, sourceName)
```

Just to make sure nobody gets confused

Reply:
The rest of the code uses `catalogName` and I'm not sure if `sourceName` is clearer than that.

Reply:
How about just "name" ?

It's `Catalog`.Remove()` what do catalogs remove? sources? what could the name be? the name of a source to remove?

Reply:
I'm not sure anyone new to the codebase would be able to deduce that `catalog.Delete(name string)` deletes a source. Nowhere in the catalog code we've used the word source. Also, this changelist did not introduce the word `catalogName`, it's already being used in other catalog operations.

Reply:
> this changelist did not introduce the word catalogName, it's already being used in other catalog operations

True, and that is a mistake given we have already conflated catalogs and sources multiple times in discussion.

Reply:
I'm happy to rename all `catalogName` references to `sourceName`. WDYT?

Reply:
yeh i think that would help unmuddle things 😅 , thanks!